### ISSUE TITLE
UuidBinaryModelTrait error when using toArray or toJSON

### ISSUE THREAD

**Issue by cosmini** [alsofronie/eloquent-uuid#30]
I was getting the following error:
>PHP warning:  preg_match() expects parameter 2 to be string, object given in /home/.../vendor/alsofronie/eloquent-uuid/src/UuidBinaryModelTrait.php on line 105

I had to override the method in my model by adding extra check if the value is not object:
```
private function deepArray($array)
    {
        foreach ($array as $key => $value) {
            $value = $array[$key];
            if (is_array($value)) {
                $array[$key] = $this->deepArray($value);
            } elseif (!is_object($value) && !preg_match('//u', $value)) {
                // TODO: drop the preg_match because it's slow and
                // what if a binary value in the uuid gets represented
                // by valid ASCII or UTF symbols?
                $array[$key] = (property_exists($this, 'uuidOptimization') && $this::$uuidOptimization) ?
                    self::toNormal($value) :
                    bin2hex($value);
            }
        }

        return $array;
    }
```

**Comment by alsofronie** [comment]
Hi. I can't reproduce the error. Can you provide a context?

**Comment by cosmini** [comment]
I added some accessors and mutators to my Laravel model.
For example I have declared the following:
`protected $appends = ['example'];`
And have implemented a method to get that attribute:
`public function getExampleAttribute() {
     return $this->created_at;
}`

This attribute is not being added into the model object, but if method `toArray` or `toJSON` is being called they will be added into the response.

I suggest that you have a test by appending an extra attribute to your model.
